### PR TITLE
Add time range to stats 

### PR DESCRIPTION
**Pull request by gadotroee**

### CODE REVIEW

File: ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
+import { TimeRangePicker } from "./TimelineBarChart/TimeRangePicker/TimeTangePicker";
 
 const modalStyle = {
   position: 'absolute',
   top: '6%',
   left: '50%',
   transform: 'translate(-50%, 0%)',
   width: '60vw',
-  height: '82vh',
+  height: '90vh',
```
Comment (leon-up9) [inline ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx:17]: why this change was necessary?

File: ui-common/package.json [hunk 2]
```diff
@@ -85,7 +86,7 @@
     "rollup-plugin-postcss": "^4.0.2",
     "rollup-plugin-sass": "^1.2.12",
     "rollup-plugin-scss": "^3.0.0",
-    "typescript": "^4.7.2"
+    "typescript": "^4.5.3"
```
Comment (leon-up9) [inline ui-common/package.json:89]: why this was changed?

File: ui/src/App.tsx [hunk 3]
```diff
@@ -20,26 +22,28 @@ const App = () => {
...
-                        getOasServices={api.getOasServices}
-                        getOasByService={api.getOasByService}
-                        openModal={oasModalOpen}
-                        handleCloseModal={() => setOasModalOpen(false)}
-                    />}
-                    <TrafficStatsModal isOpen={trafficStatsModalOpen} onClose={() => setTrafficStatsModalOpen(false)} getTrafficStatsDataApi={api.getTrafficStats}/>
-                </div>
-            </ThemeProvider>
-        </StyledEngineProvider>
+        <EuiProvider>
```
Comment (leon-up9) [inline ui/src/App.tsx:25]: why its on the whole level of the app?

File: ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx (hunk 1 above)
Comment (AmitUp9) [reply ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx:17]: on big screens the timeline is getting near the edge

File: ui/src/App.tsx (hunk 3 above)
Comment (AmitUp9) [reply ui/src/App.tsx:25]: You think its better the wrap only the component that uses the traffic stats from the ui-common?

Comment (leon-up9) [reply ui/src/App.tsx:25]: yes, if its not used by the whole app,no need

File: ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx (hunk 1 above)
Comment (leon-up9) [reply ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx:17]: lets talk about it

File: ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx [hunk 4]
```diff
@@ -26,26 +27,27 @@ const modalStyle = {
...
   const [statsMode, setStatsMode] = useState(modes[0]);
   const [selectedProtocol, setSelectedProtocol] = useState(ALL_PROTOCOLS);
   const [pieStatsData, setPieStatsData] = useState(null);
   const [timelineStatsData, setTimelineStatsData] = useState(null);
   const [protocols, setProtocols] = useState([])
   const [isLoading, setIsLoading] = useState(false);
-  const commonClasses = useCommonStyles();
 
-  const getTrafficStats = useCallback(async () => {
+  const now = new Date().getTime();
```
Comment (lirazyehezkel) [inline ui-common/src/components/modals/TrafficStatsModal/TrafficStatsModal.tsx:42]: move into useEffect

File: ui-common/src/components/modals/TrafficStatsModal/TimelineBarChart/TimeRangePicker/TimeTangePicker.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    const [isLoading, setIsLoading] = useState(false);
+    const [start, setStart] = useState('now-30m');
+    const [end, setEnd] = useState('now');
+    const [isPaused, setIsPaused] = useState(true);
+    const [refreshInterval, setRefreshInterval] = useState();
+
+    const dateConvertor = (inputStart, inputEnd) => {
+        const startMoment = dateMath.parse(inputStart);
+        if (!startMoment || !startMoment.isValid()) {
+            throw new Error('Unable to parse start string');
```
Comment (lirazyehezkel) [inline ui-common/src/components/modals/TrafficStatsModal/TimelineBarChart/TimeRangePicker/TimeTangePicker.tsx:24]: where do you catch this error?

Comment (AmitUp9) [reply ui-common/src/components/modals/TrafficStatsModal/TimelineBarChart/TimeRangePicker/TimeTangePicker.tsx:24]: I think we can replace it with console.error...what do you think?

File: ui-common/src/components/modals/TrafficStatsModal/TimelineBarChart/TimeRangePicker/TimeTangePicker.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                : recentlyUsedRange
+        );
+        const { startMoment, endMoment } = dateConvertor(start, end)
+        refreshStats(startMoment, endMoment)
+        setIsLoading(true);
+        startLoading();
+    };
+
+    const onRefresh = ({ start, end, refreshInterval }) => {
+        return new Promise(resolve => {
```
Comment (lirazyehezkel) [inline ui-common/src/components/modals/TrafficStatsModal/TimelineBarChart/TimeRangePicker/TimeTangePicker.tsx:54]: why do you need Promise?

Comment (AmitUp9) [reply ui-common/src/components/modals/TrafficStatsModal/TimelineBarChart/TimeRangePicker/TimeTangePicker.tsx:54]: its the code from elastic, i didnt changed it

**Review by lirazyehezkel — APPROVED** [review]
